### ISSUE TITLE
[Feature request] Some options for analog.clock

### ISSUE THREAD

**Issue by Crystal4276** [elParaguayo/qtile-extras#206]
The final goal would be to get rid of the clock icon displayed inside widget.Clock and replace it by analog.clock.

I would need a few additional options to achieve this goal.
This is what i could get so far :
![image](https://user-images.githubusercontent.com/27446196/204134970-82aafb87-77a3-40b5-a0b3-22a7ed6cbe4d.png)

Additional options:
- minute_colour
- hour_colour
- margin parameter splitted in margin_x and margin_y (i'm using margin to tune the widget size. But this introduce extra space on x-axis) 
- offset_y (tune the position of the clock on y-axis to match text position)

```python
                widget.AnalogueClock(
                    face_shape='circle',
                    face_border_width=3,
                    margin=20,
                    # ~ padding=20,
                    hour_length=0.45,
                    minute_length=0.7,
                    background=colors[7], 
                    minute_colour=colors[10],
                    face_border_colour=colors[10],
                    **decor_general, 
                ),
                widget.Clock( 
                       font="Symbols Nerd Font Mono",
                       padding = 0,
                       foreground = colors[10],
                       background=colors[7],
                       fontsize = 26,
                       format="  %H:%M",
                       **decor_general, 
                ),
```

**Comment by elParaguayo** [comment]
I'll take a look. Should be easiest enough.

Re y offset, it is centered vertically. I'd argue that your icon in the screenshot is too low!

**Comment by Crystal4276** [comment]
> Re y offset, it is centered vertically. I'd argue that your icon in the screenshot is too low!

Yeah i agree. But i don't have this flexibility with other widgets either.
Those NvidiaSensors, CPU, Memory, Clock, TextBox widgets do not position their content in the center of the bar:
![image](https://user-images.githubusercontent.com/27446196/204135508-80404d24-1636-49f0-b773-c71d3802b647.png)
And they don't have an offset_y option to correct it :(

**Comment by elParaguayo** [comment]
Should hopefully work now.

**Comment by Crystal4276** [comment]
Thanks !
Color, adjust_x/y are working.

I still have an issue with the left and right space around the widget
![image](https://user-images.githubusercontent.com/27446196/204155102-dd114191-8ddc-4153-80a1-1f77e34032ff.png)

Just a guess (trying to understand how it works), maybe this line 80:
```python
self.length = self.bar.size + 2 * self.padding
```
It looks like the effect of margin is not taken into account.

My config:
```python
 widget.AnalogueClock(
                    margin=20,
                    padding=0,
                    hour_size= 2,
                    hour_length=0.45,
                    minute_size= 2,
                    minute_length=0.7,
                    background=colors[7], 
                    hour_colour=colors[10],
                    minute_colour=colors[10],
                    face_border_colour=colors[10],
                    face_shape='circle',
                    face_border_width=3,
                    update_interval=1,
                    **decor_general, 
                ),

decor_general = {
    "decorations": [
        RectDecoration(
            #colour="#a6e3a1",
            use_widget_background=True,
            line_width= 1,
            line_colour=colors[17],
            radius=[15, 15, 15, 15],
            filled=True,
            margin_y=20,
            padding_y=6,
            padding_x=0,
            group=True,
            clip=True,
        ),
    ],
}
```

**Comment by elParaguayo** [comment]
I can't quite tell from that picture what your issue is.

Anyway, I've added the `margin` to the widget's length so you can test and see if it does what you're after.

**Comment by Crystal4276** [comment]
Sorry i was not explicit enough.
I need the space between the border of Rectdecoration and the clock to be removed.
![image](https://user-images.githubusercontent.com/27446196/204156196-f1633005-67e3-4ff5-9b09-e6d643fed1ae.png)

If I modify this
line 80 ` self.length = self.bar.size + 2 * (self.padding + self.margin)`
by this:  ` self.length = self.bar.size + 2 * (self.padding - self.margin)`
Then i get this:
![image](https://user-images.githubusercontent.com/27446196/204156331-27d73191-fefe-42ad-b487-0992493d6f31.png)
Almost there !
If i correct the position on x axis by substraction the margin
```python
margin =18
adjust_x=-18
```
Then i get what i'm looking for:
![image](https://user-images.githubusercontent.com/27446196/204156454-b79bb72a-157b-49de-a94d-d4f024a085dc.png)

To summarize:
- if we can rather modify on line 80:  ` self.length = self.bar.size + 2 * (self.padding - self.margin)`
- the margin value should be subtracted somewhere for the x position ( i can't tell where )

**Comment by elParaguayo** [comment]
I know you've set `padding=0` in your config for this widget but do you also have it in your `widget_defaults`?

**Comment by Crystal4276** [comment]
I don't have widget_defaults defined (i deleted it).
Is that an issue ?

**Comment by elParaguayo** [comment]
No but it may override the value you set so I just wanted to check that the size issue wasn't related to that.
